sqlmesh/core/context.py

```diff
@@ -1644,13 +1654,22 @@ def plan_builder(
         elif forward_only is None:
             forward_only = self.config.plan.forward_only
 
+        # When handling prod restatements, only clear intervals from other model versions if we are using full virtual environments
+        # If we are not, then there is no point, because none of the data in dev environments can be promoted by definition
+        clear_restated_intervals_across_model_versions = (
```

Comment:
`versions` seems overloaded here. I initially interpreted this as a snapshot version, when in reality this means model version aka "snapshots". Should we call this `restate_all_snapshots`?

Reply:
> It's also not all snapshots - just ones that arent promoted in prod.

well, the prod ones are cleared by definitions, they just get immediately backfilled. The fact that we don't update state for those is an implementation detail and is not relevant here.